### PR TITLE
Add test utilities (Jest snapshot serialisers)

### PR DESCRIPTION
**Pull request by andreypopp**
A new module `react-stylesheet/testutils` which exposes just the
snapshot serializers for Jest.

The example test setup looks like this:

    import React from 'react';
    import renderer from 'react-test-renderer';

    import {Element} from 'react-stylesheet';
    import * as TestUtils from 'react-stylesheet/testutils';

    expect.addSnapshotSerializer(TestUtils.snapshotSerializer);

    function Hello() {
      return <Element color="red" colorOnHover="black">HEllo!</Element>
    }

    test('rendering <Hello />', function() {
      const tree = renderer.create(<Hello />).toJSON();
      expect(tree).toMatchSnapshot();
    });

Which produces the following snapshot:

    <div
      className={
        StyleJoin [
          PrecompiledCSS {
            "boxSizing": "border-box",
          },
          DynamicallyGeneratedCSS {
            "hover": Object {
              "color": "black",
            },
          },
        ]
      }
      style={Object {
        "hover": "red",
      }}
    >
      Hello
    </div>

This is for `<Element />`, styled components have a little bit different
representation (you can take a look into updated/new snapshots to get
the idea).

Few other observations:

* We only make it work when `process.env.NODE_ENV === "test"` because
keeping the original style info means we need to allocate wrappers for
CSS classes which could be costly at runtime.

### CODE REVIEW

File: src/TestUtils.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export const snapshotSerializer = {
+  // $ExpectError: ...
+  print(val, serialize, _indent) {
+    if (val instanceof CSSClassRepr) {
+      const repr = serialize(val.repr);
+      return repr.replace(/^Object {/, `${val.reprName} {`);
+    } else if (val instanceof CSSClassJoinRepr) {
+      const repr = serialize(val.styles);
+      return repr.replace(/^Array \[/, 'StyleJoin [');
```
Comment (andreypopp) [inline src/TestUtils.js]: Hacky but works!

File: src/TestUtils.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+import {CSSClassRepr, CSSClassJoinRepr} from './CSSClassRepr';
+
+export const snapshotSerializer = {
+  // $ExpectError: ...
+  print(val, serialize, _indent) {
+    if (val instanceof CSSClassRepr) {
+      const repr = serialize(val.repr);
+      return repr.replace(/^Object {/, `${val.reprName} {`);
```
Comment (andreypopp) [inline src/TestUtils.js]: Hacky but works!

File: src/CSSClassRepr.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/**
+ * @flow
+ */
+
+export class CSSClassRepr {
+  className: string;
+  repr: any;
```
Comment (lapanoid) [inline src/CSSClassRepr.js]: what is repr means?

Comment (lapanoid) [inline src/CSSClassRepr.js]: If this is "representation", can we call it fullname? And CSSClassRepresentation? Maybe it is long to type, but more obvious

**Review by sergey-lapin — APPROVED** [review]

**Comment by andreypopp** [comment]
This is going to be a 3.0.0 release as it will invalidate stored test snapshots for existing users.

**Pull request merged**